docs/csharp/language-reference/operators/snippets/patterns/DeclarationAndTypePattern.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Patterns
+{
+    public static class DeclarationAndTypePattern
+    {
+        public static void Examples()
+        {
+            ReferenceConversion();
+            NullableAndUnboxing();
+        }
+
+        private static void ReferenceConversion()
+        {
+            // <ReferenceConversion>
+            var numbers = new int[] { 10, 20, 30 };
+            Console.WriteLine(GetSourceLabel(numbers));  // output: 1
+
+            var letters = new List<char> { 'a', 'b', 'c', 'd' };
+            Console.WriteLine(GetSourceLabel(letters));  // output: 2
+
+            static int GetSourceLabel<T>(IEnumerable<T> source) => source switch
+            {
+                Array a => 1,
+                ICollection<T> c => 2,
```

Comment:
```suggestion
                Array array => 1,
                ICollection<T> collection => 2,
```

I think it was fine to have simple variable names `a`, and `c`, but I believe it drives home the desire to add the type pattern features home a bit more. Having a verbose variable name just makes the argument that much more satisfying.